### ISSUE TITLE
wrong error text when multiple positional arguments are expected

### ISSUE THREAD

**Issue by tfki** [p-ranav/argparse#208]
if multiple positional arguments are added like so
`argparse::ArgumentParser parser{ "my program" };
  parser.add_argument("positional1").help("first positional argument");
  parser.add_argument("positional2").help("second positional argument");`

and the number of arguments passed to the program is lower than, in this case, two, "1 argument(s) expected. 0 provided." is output. It does not matter whether the arguments are `.required()` or not. In this case i would expect it to say "2 argument(s) expected[...]" and the correct amount of arguments provided, it does neither. To me this seems like an error.

**Comment by skrobinson** [comment]
I agree that you've found a bug. The confusing output is because `positional1` expects one argument and immediately throws, so no parsing is done for `positional2`. This is how `argparse` works, but the error message could be more clear.
